vendor/parser-jane/for-parser-standard/parser.mly

```diff
@@ -153,9 +153,31 @@ let mk_attr ~loc name payload =
   Attr.mk ~loc name payload
 
 let mkpat_with_modes ~loc ~pat ~cty ~modes =
-  match cty, modes with
-  | None, [] -> pat
-  | cty, modes -> mkpat ~loc (Ppat_constraint (pat, cty, modes))
+  match pat.ppat_desc with
```

Comment:
Does this check need to be recursive?  Could I ever contrive to have several nested pat constraint nodes with only modes?  (I weakly think no, because modes are allowed are somewhat restricted, but you will have a better sense)

Reply:
The only places where a `Ppat_constraint` is constructed outside this function in the parser all have no modes, so I don't think recursion is needed.